packages/ember-data/tests/integration/serializers/rest_serializer_test.js

```diff
@@ -284,19 +284,6 @@ test("serialize polymorphicType", function() {
   });
 });
 
-test("serialize polymorphicType with decamelized typeKey", function() {
-  YellowMinion.typeKey = 'yellow-minion';
-  var tom, ray;
-  run(function() {
-    tom = env.store.createRecord(YellowMinion, { name: "Alex", id: "124" });
-    ray = env.store.createRecord(DoomsdayDevice, { evilMinion: tom, name: "DeathRay" });
-  });
-
-  var json = env.restSerializer.serialize(ray._createSnapshot());
-
-  deepEqual(json["evilMinionType"], "yellowMinion");
-});
-
```

Comment:
I believe this test is incorrect.

* You should never manually set a `typeKey`.
* The `YellowMinion` was registered into the container with camelCase.
* This is a unit test, there is no normalization of model names

Looking into polymorphics, I am confused by the type usage. Sometimes it seems like we trust the typeKey ([serialization](https://github.com/emberjs/data/blob/f71e90a76f8290f98315a1776a5d158b9c463694/packages/ember-data/lib/serializers/rest_serializer.js#L726)) other times we camelize it ([relationType](https://github.com/emberjs/data/blob/f71e90a76f8290f98315a1776a5d158b9c463694/packages/ember-data/lib/serializers/rest_serializer.js#L746)).